src/lib.rs

```diff
@@ -74,28 +67,9 @@ impl AsRef<[u8]> for OutgoingCipherKey {
 /// Newtype representing the byte encoding of an [`EphemeralPublicKey`].
 ///
 /// [`EphemeralPublicKey`]: Domain::EphemeralPublicKey
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub struct EphemeralKeyBytes(pub [u8; 32]);
 
-impl fmt::Debug for EphemeralKeyBytes {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        struct HexFmt<'b>(&'b [u8]);
-        impl<'b> fmt::Debug for HexFmt<'b> {
-            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                f.write_char('"')?;
-                for b in self.0 {
-                    f.write_fmt(format_args!("{:02x}", b))?;
-                }
-                f.write_char('"')
-            }
-        }
-
-        f.debug_tuple("EphemeralKeyBytes")
-            .field(&HexFmt(&self.0))
-            .finish()
-    }
-}
```

Comment:
why this was removed?

Reply:
It turns out this change was made in the zcash/librustzcash PR https://github.com/zcash/librustzcash/pull/926 by @str4d, which was merged on 2023-09-01. However, we only merged zcash/librustzcash changes into our QED-it/librustzcash fork until 2023-06-06. So, I guess we can upgrade our QED-it/librustzcash from zcash/librustzcash upstream, and then I will recopy zcash_note_encryption from QED-it/librustzcash here and recreate the PR. That would allow us to track the changes correctly.

Reply:
Done. Instead of implementing the suggestion above, I cherry-picked the missed commit for `zcash_note_encryption` from `zcash/librustzcash`.